haystack/components/converters/csv.py

```diff
@@ -102,3 +105,233 @@ def run(
             documents.append(document)
 
         return {"documents": documents}
+
+
+@component
+class CSVSplitter:
```

Comment:
I'd advocate for changing the name of the component to `CSVDocumentSplitter` to be line with our naming of our other splitters to emphasize it works on Haystack Documents.